mythril/laser/ethereum/natives.py

```diff
@@ -3,36 +3,40 @@
 import copy
 import hashlib
 import logging
+from typing import Union
 
 from ethereum.utils import ecrecover_to_pub
 from py_ecc.secp256k1 import N as secp256k1n
 from rlp.utils import ALL_BYTES
 
 from mythril.laser.ethereum.util import bytearray_to_int, sha3
+from mythril.laser.ethereum.state import Calldata
 
 
 class NativeContractException(Exception):
     pass
 
 
-def int_to_32bytes(i):  # used because int can't fit as bytes function's input
+def int_to_32bytes(
+    i: int
+) -> bytes:  # used because int can't fit as bytes function's input
     o = [0] * 32
     for x in range(32):
         o[31 - x] = i & 0xFF
         i >>= 8
     return bytes(o)
 
 
-def extract32(data, i):
+def extract32(data: bytearray, i: int) -> int:
     if i >= len(data):
         return 0
     o = data[i : min(i + 32, len(data))]
     o.extend(bytearray(32 - len(o)))
     return bytearray_to_int(o)
 
 
-def ecrecover(data):
-
+def ecrecover(data: Union[bytes, str]) -> bytes:
```

Comment:
can you add a list too into the `data`'s type?

Reply:
bunch of ascii numbers from what i saw.

Reply:
Done :+1:

Reply:
I'm not seeing a change here, is that correct?